### ISSUE TITLE
Kingdom Under Fire Hero's (Assertion + Water Texture Bug)

### ISSUE THREAD

**Issue by FJCFJC123** [xemu-project/xemu#1356]
### Title

https://xemu.app/titles/504c0006/#Kingdom-Under-Fire-Heroes

### Bug Description

There is two assertion messages that will appear when leaving the main menu screen, to reproduce simply need to start a campaign and begin a mission, the start of each mission has a mission briefing which is where the rolling text begins assertions will happen right before you reach this screen. You can also trigger the assertions by letting the game sit at the main menu for a min starting a cutscene due to inactivity. You can ignore both assertions and gameplay will resume normally 

Separate issue is when water is rendered the textures are not seamless creating a square patterns

![20230108_173337](https://user-images.githubusercontent.com/86930879/213027900-5a20f51c-5f10-4ec5-bc8b-5e198fe997ef.jpg)
![20230108_173342](https://user-images.githubusercontent.com/86930879/213027904-5a2f0c7e-6879-4fa0-bdec-eeb31d08cb2a.jpg)
![xemu-2023-01-08-17-45-24](https://user-images.githubusercontent.com/86930879/213027948-6bb5eca7-7c13-431d-86c4-a1fefc2ffc7a.png)
![xemu-2023-01-08-17-45-22](https://user-images.githubusercontent.com/86930879/213027952-4a9e3bca-886b-4c42-8de7-56b615d2b4a6.png)

### Expected Behavior

Expected behavior would be no assertion's, and no square patterning on the water  

### xemu Version

Tested on Version 0.7.72

### System Information

Windows 10
i5 8600k
GTX 1080 8GB
64GB DDR4 3200HZ Ram

### Additional Context

does not effect game in terms of playability can still complete campaigns and such as normal

**Comment by abaire** [comment]
The water may be #57 as it looks very similar to the failure mode in Morrowind.

**Comment by FJCFJC123** [comment]
I figured id give it a retest cause its been a bit, Open GL now crashes if you decide to ignore the GL Framebuffer Assert previously you could bypass the error, Vulkan can has no issue at the same point, and the water bug also appears on Vulkan with the same patterning

**Comment by FJCFJC123** [comment]
Mistake closing out issue, but in retrospect i probably should of had them has separate issues im assuming.

**Comment by abaire** [comment]
Yeah, they should ideally be separate issues. I'm relatively confident from your screenshot that the water issue is the same as Morrowind; the heightmap manipulation applies compounding errors due to noise along the edges that would be rounded off in HW but is not in xemu today.


Unfortunately those assertions don't give actionable info without running under a debugger or enabling GL debugging, so this will need to be reproduced and investigated in more detail.

As an aside: in general you shouldn't bypass xemu assert messages as any behavior after a failed assert is going to be potentially undefined. It can lead to a lot of confusion as the emulated machine can be in what should be an impossible state, possibly causing all kinds of wacky issues.

**Comment by FJCFJC123** [comment]
Right on when, ill separate the issues and get them updated to git and close this out either tonight or tomorrow, but yeah, just wasnt sure if that maybe valuable to yall in the debugging sense that at one point it would just throw those two Frame Complete and No Error asserts and carry on, and now it throws a string of asserts following the bypass of the original two.

**Comment by FJCFJC123** [comment]
![Image](https://github.com/user-attachments/assets/908356f7-675d-409d-99d3-7a6a0f267a61)   ill leave this here for now too, from my test with vulkan last night a bit of a close up of the water.

**Comment by abaire** [comment]
I can't reproduce the GL errors on M3 macOS w/ 0.8.55 or on Windows 10, GTX1070 NVIDIA drivers 572.70 w/ xemu 0.8.58, but maybe I'm missing something wrt. how to reproduce?

I just loaded the game and kept hitting A to create a new game, select the default character, then proceed past the loading screen and into the actual game. 

Is there a step that I've missed?

**Comment by FJCFJC123** [comment]
Interesting, pretty much the trigger point was right after starting/loading a campagin it would have a loading screen with some scrolling mission details. I will double check my dump to make sure its not something on my end although i feel the dumps okay cause vulkans running fine, once im off work today ill report back.

**Comment by FJCFJC123** [comment]
My system specs are the same as the original report with the only change being a 2080ti instead of the 1080

**Comment by FJCFJC123** [comment]
Sorry for the delayed response, appears that GL issue was a weird one off, retested and it kicked the error, so i decided to swap off cerbios to complex _4627 which then the GL asserts were not happening, so i then swapped back to cerbios and now it loads without issue on cerbios (so im gonna say its safe to assume that was somehow tethered locally to my setup), so im gonna close this out and open up a new issue just for the water bug.
